media/js/firefox/new/experiment-firefox-new-waitface.js

```diff
@@ -0,0 +1,46 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+
+/* Experiment: https://bugzilla.mozilla.org/show_bug.cgi?id=1434764
+ * URL: https://www.mozilla.org/en-US/firefox/new/?xv=waitface
+ * Audience: Desktop, Non-Fx, en-US with utm_medium=display&utm_source=doubleclick    
+ * Traffic: 100% audience, 50/50 split*/
+
+(function() {
+    'use strict';
+
+    var isDesktop = function() {
```

Comment:
We have a [client detection library](https://github.com/mozilla/bedrock/blob/master/media/js/base/mozilla-client.js) in our common page bundle for convenience. You should be able to do the following:

```javascript
var client = Mozilla.Client;

if (client.isDesktop && !client.isFirefox && paramsMatch()) {
    // run experiment
}
```

Reply:
Regarding this and other questions like it: Mozilla.Client and many other utilities are loaded at the bottom of the document by a base template, and TC is loaded at the top, and those both seem like good things. Rather than change one of those (which might affect perceived performance), @jpetto and I agreed I should just pull out the relevant bits. But I can change it easily if you think I should.

Reply:
Ah, yes - I neglected to consider where in the document this file is loaded. I think what you've done makes sense.